homeassistant/components/comelit/climate.py

```diff
@@ -0,0 +1,206 @@
+    HVACMode.HEAT: ClimaAction.MANUAL,
+}
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up Comelit climates."""
+
+    coordinator: ComelitSerialBridge = hass.data[DOMAIN][config_entry.entry_id]
+
+    async_add_entities(
+        ComelitClimateEntity(coordinator, device, config_entry.entry_id)
+        for device in coordinator.data[CLIMATE].values()
+    )
+
+
+class ComelitClimateEntity(CoordinatorEntity[ComelitSerialBridge], ClimateEntity):
+    """Climate device."""
+
+    _attr_hvac_modes = [HVACMode.AUTO, HVACMode.COOL, HVACMode.HEAT, HVACMode.OFF]
+    _attr_max_temp = 30
+    _attr_min_temp = 5
+    _attr_supported_features = ClimateEntityFeature.TARGET_TEMPERATURE
+    _attr_target_temperature_step = PRECISION_TENTHS
+    _attr_temperature_unit = UnitOfTemperature.CELSIUS
+    _attr_has_entity_name = True
+    _attr_name = None
+
+    def __init__(
+        self,
+        coordinator: ComelitSerialBridge,
+        device: ComelitSerialBridgeObject,
+        config_entry_entry_id: str,
+    ) -> None:
+        """Init light entity."""
+        self._api = coordinator.api
+        self._device = device
+        super().__init__(coordinator)
+        # Use config_entry.entry_id as base for unique_id
+        # because no serial number or mac is available
+        self._attr_unique_id = f"{config_entry_entry_id}-{device.index}"
+        self._attr_device_info = coordinator.platform_device_info(device, device.type)
+
+    @property
+    def _clima(self) -> list[Any]:
+        """Return clima device data."""
+        # CLIMATE has 2 turple:
+        # - first  for Clima
+        # - second for Humidifier
+        return self.coordinator.data[CLIMATE][self._device.index].val[0]
+
+    @property
+    def _api_mode(self) -> str:
+        """Return device mode."""
+        # Values from API: "O", "L", "U"
+        return self._clima[2]
+
+    @property
+    def _api_active(self) -> bool:
+        "Return device active/idle."
+        return self._clima[1]
+
+    @property
+    def _api_automatic(self) -> bool:
+        """Return device in automatic/manual mode."""
+        return self._clima[3] == ClimaMode.AUTO
+
+    @property
+    def target_temperature(self) -> float:
+        """Set target temperature."""
+        return self._clima[4] / 10
+
+    @property
+    def current_temperature(self) -> float:
+        """Return current temperature."""
+        return self._clima[0] / 10
+
+    @property
+    def hvac_mode(self) -> HVACMode | None:
+        """HVAC current mode."""
+
+        if self._api_mode == ClimaMode.OFF:
+            return HVACMode.OFF
+
+        if self._api_automatic:
+            return HVACMode.AUTO
+
+        if self._api_mode in API_STATUS:
+            return API_STATUS[self._api_mode]["hvac_mode"]
+
+        _LOGGER.warning("Unknown API mode '%s' in hvac_mode", self._api_mode)
```

Comment:
We don't want side effects like logging in entity state properties. Please remove it or move it.